src/util/simplify_expr.cpp

```diff
@@ -1744,6 +1744,31 @@ simplify_exprt::simplify_byte_extract(const byte_extract_exprt &expr)
     }
   }
 
+  // byte_extract(byte_update(root, offset_u, value), offset_e) so that the
+  // extracted range fully lies within the update value simplifies to
+  // byte_extract(value, offset_e - offset_u)
+  if(
+    expr.op().id() == ID_byte_update_big_endian ||
```

Comment:
Quick question: Why not `can_cast_expr<byte_update_exprt>(expr.op())` from `src/util/byte_operators.h`?

It seems that it does the same operation, and makes sense to be the canonical place for handling the knowledge of what is a `byte_update_expr`, so it would make sense to reuse that. What do you think?

Reply:
Good call, thank you!